src/app/annotationViewer/positionLine.js

```diff
@@ -5,121 +5,123 @@ angular
+                };
 
                 /**
                  * converts the proportion of elapsed audio to a pixel value,
-                 * based on the width of the spectrogram image.
+                 * based on the width of the containing element of the position line, which should match the width of the spectrogram image.
                  * @param audioPositionSeconds number
                  * @returns number
-                 * @TODO: remove this. Better to use the unit converter, and also better if we don't need to reference the dom element
+                 * @TODO: can we use the unit converter for this?
                  */
-                scope.secondsToPixels = function (audioPositionSeconds) {
-                    var pixelPosition, imageWidth;
-
-                        if (image.clientWidth > 0) {
-                            imageWidth = image.clientWidth;
-                        } else {
-                            // use number of columns of spectrogram as image width
-                            imageWidth = 1;
-                        }
-
-                        pixelPosition = imageWidth * (scope.secondsToRatio(audioPositionSeconds));
-
-                    return pixelPosition;
+                self.secondsToPixels = function (audioPositionSeconds) {
+                    var width = self.getWidth() * (self.secondsToRatio(audioPositionSeconds));
+                    return Math.min(width, self.getWidth());
                 };
 
+                self.pixelsToSeconds = function (numPixels) {
+                    return (numPixels / self.getWidth()) * self.audioDuration();
+                };
 
                 /**
                  * Converts a number of seconds of elapsed audio to the fraction of the audio that has elapsed
                  * by dividing by the duration of the audio.
                  * @param audioPositionSeconds number
                  * @returns {number} range [0,1]
                  */
-                scope.secondsToRatio = function (audioPositionSeconds) {
-                    if (scope.media && scope.media.endOffset) {
-                        var positionRatio = audioPositionSeconds / (scope.media.endOffset - scope.media.startOffset);
-                        return (positionRatio > 1) ? 1 : positionRatio;
-                    }
-                    return 0;
+                self.secondsToRatio = function (audioPositionSeconds) {
+                    var positionRatio = audioPositionSeconds / self.audioDuration();
+                    return Math.min(1, positionRatio);
+                };
 
+                self.ratioToSeconds = function (ratio) {
+                    return self.audioDuration() * ratio;
                 };
 
+                self.audioDuration = function () {
+                    if (scope.media && scope.media.endOffset) {
+                        return scope.media.endOffset - scope.media.startOffset;
+                    } else {
+                        return 0.001;
```

Comment:
Changed to return 0. It was like that to prevent dividing by zero. This is now checked where the division is done.